### PR TITLE
add bucket feature

### PR DESCRIPTION
**Pull request by VictoriqueMoe**
add buckets feature.

### CODE REVIEW

File: src/model/db/FileUpload.model.ts [hunk 1]
```diff
@@ -87,6 +88,20 @@ export class FileUploadModel extends AbstractModel {
     })
     public encrypted: boolean;
 
+    @Column({
+        nullable: true,
+    })
+    public bucketToken: string | null;
+
+    @ManyToOne("BucketModel", "submissions", {
```
Comment (VictoriqueMoe) [inline src/model/db/FileUpload.model.ts:96]: `submissions` is wrong, this should be `files`

Comment (nakedmcse) [reply src/model/db/FileUpload.model.ts:96]: Updated in my local copy

Comment (nakedmcse) [reply src/model/db/FileUpload.model.ts]: Pushed change

**Comment by nakedmcse** [comment]
First Draft of bucket admin pages basically works, but has some issues to be fixed:

1.  Files don't show on first load - on changing page to stats and back then they do.
2. Stats page needs filtered

I removed access to the user password page, and to ban/unban IP when using a bucket token.  Verified both still work when logged in normally.

This commit also contains a fix to createBucket - it would infinite loop when files were empty, which they are when you create a bucket.

Tested here with three files uploaded with a bucket and three not.  Logged in normal shows 6, logged in using the bucket ID shows 3.

File: src/model/dto/BucketDto.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ export class BucketDto {
     public files: FileUploadResponseDto[];
 
     public static fromModel(model: BucketModel, baseUrl: string): BucketDto {
-        const fileDtos = model.files.map(f => FileUploadResponseDto.fromModel(f, baseUrl));
+        const fileDtos = model.files ? model.files.map(f => FileUploadResponseDto.fromModel(f, baseUrl)) : [];
```
Comment (VictoriqueMoe) [inline src/model/dto/BucketDto.ts:19]: why? if tghe bucket is empty, the files array is empty. so why are you checking if files is nullish? can fileds even be nullish?

**Review by nakedmcse — APPROVED** [review]
Approved

**Pull request merged**